docs-ref-conceptual/eclipse/java-sdk-azure-get-started-eclipse.md

```diff
@@ -0,0 +1,415 @@
+- [Eclipse](http://www.eclipse.org/downloads/)
+
+## Set up authentication
+
+Your Java application needs read and create permissions in your Azure subscription to run the sample code in this tutorial. Create a service principal and configure your application to run with its credentials. Service principals provide a way to create a non-interactive account associated with your identity to which you grant only the privileges your app needs to run.
+
+[Create a service principal using the Azure CLI 2.0](/cli/azure/create-an-azure-service-principal-azure-cli) and capture the output. Provide a [secure password](https://docs.microsoft.com/azure/active-directory/active-directory-passwords-policy) in the password argument instead of `MY_SECURE_PASSWORD`. Your password must be 8 to 16 characters and match at least 3 out of the 4 following criteria:
+
+* Include lowercase characters
+* Include uppercase characters
+* Include numbers
+* Include one of the following symbols: @ # $ % ^ & * - _ ! + = [ ] { } | \ : ‘ , . ? / ` ~ “ ( ) ;
+
+
+```azurecli-interactive
+az ad sp create-for-rbac --name AzureJavaTest --password "MY_SECURE_PASSWORD"
+```
+
+Which gives you a reply in the following format:
+
+```json
+{
+  "appId": "a487e0c1-82af-47d9-9a0b-af184eb87646d",
+  "displayName": "AzureJavaTest",
+  "name": "http://AzureJavaTest",
+  "password": password,
+  "tenant": "tttttttt-tttt-tttt-tttt-tttttttttttt"
+}
+```
+
+Next, copy the following into a text file on your system:
+
+```text
+# sample management library properties file
+subscription=ssssssss-ssss-ssss-ssss-ssssssssssss
+client=cccccccc-cccc-cccc-cccc-cccccccccccc
+key=kkkkkkkkkkkkkkkk
+tenant=tttttttt-tttt-tttt-tttt-tttttttttttt
+managementURI=https\://management.core.windows.net/
+baseURL=https\://management.azure.com/
+authURL=https\://login.windows.net/
+graphURL=https\://graph.windows.net/
+```
+
+Replace the top four values with the following:
+
+- subscription: use the *id* value from `az account show` in the Azure CLI 2.0.
+- client: use the *appId* value from the output taken from a service principal output.
+- key: use the *password* value from the service principal output.
+- tenant: use the *tenant* value from the service principal output.
+
+Save this file in a secure location on your system where your code can read it. You may use this file for future code so it's recommended to store it somewhere external to the application in this article.
+
+Set an environment variable `AZURE_AUTH_LOCATION` with the full path to the authentication file in your shell.
+
+```bash
+export AZURE_AUTH_LOCATION=/Users/raisa/azureauth.properties
+```
+
+If you're working in a windows environment, then you want to add the variable to your system properties. Open powershell and, after replacing the second variable with the path to your file, enter the following command:
```

Comment:
If you're working in a Windows environment, add the variable to your system properties.

Reply:
PowerShell